<body>
  <main id="root"></main>
  <script src="https://unpkg.com/react@18.2.0/umd/react.development.js"></script>
  <script src="https://unpkg.com/react-dom@18.2.0/umd/react-dom.development.js"></script>
  <script src="https://unpkg.com/react-error-boundary@1.2.5/dist/umd/react-error-boundary.js"></script>
  <script src="https://unpkg.com/@babel/standalone@7.8.3/babel.js"></script>
  <script type="text/babel">
    const ErrorBoundary = ReactErrorBoundary.ErrorBoundary
    // class ErrorBoundary extends React.Component {
    //   state = { error: null }
    //   static getDerivedStateFromError(error) {
    //     return { error }
    //   }

    //   render() {
    //     const { error } = this.state
    //     if (error) {
    //       return <this.props.FallbackComponent error={error} />
    //     }
    //     return this.props.children
    //   }
    // }

    function ErrorFallback({ error }) {
      return (
        <section>
          <p>Something went wrong:</p>
          <pre>{error.message}</pre>
        </section>
      )
    }

    function Bomb() {
      throw new Error('ðŸ’¥ KABOOM ðŸ’¥')
    }

    function App() {
      const [explode, setExplode] = React.useState(false)
      return (
        <div>
          <div>
            <button onClick={() => setExplode(true)}>ðŸ’£</button>
            </div>
            <ErrorBoundary FallbackComponent={ErrorFallback}>
            <div>{explode ? <Bomb /> : 'Push the button Max!'}</div>
          </ErrorBoundary>
        </div>
      )
    }

    const root = ReactDOM.createRoot(document.getElementById('root'))
    root.render(<App />)
  </script>
</body>
